File: backend/src/main/java/ma/investpro/config/OpenApiConfig.java
package ma.investpro.config;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OpenApiConfig {

    @Bean
    public OpenAPI investProOpenAPI() {
        return new OpenAPI()
            .info(new Info()
                .title("InvestPro Maroc API")
                .description("API pour la gestion des dépenses d'investissement et calcul des commissions d'intervention")
                .version("1.0.0")
                .contact(new Contact()
                    .name("InvestPro Support")
                    .email("support@investpro.ma"))
                .license(new License()
                    .name("Propriétaire")
                    .url("https://investpro.ma")));
    }
}

File: backend/src/main/kotlin/ma/investpro/config/OpenApiConfig.kt
package ma.investpro.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import io.swagger.v3.oas.models.security.SecurityRequirement
import io.swagger.v3.oas.models.security.SecurityScheme
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfig {

    @Bean
    fun customOpenAPI(): OpenAPI {
        val securitySchemeName = "bearerAuth"

        return OpenAPI()
            .info(
                Info()
                    .title("InvestPro Maroc API")
                    .version("1.0.0")
                    .description("API pour la gestion des dépenses d'investissement et calcul des commissions")
                    .contact(
                        Contact()
                            .name("InvestPro Team")
                            .email("contact@investpro.ma")
                    )
                    .license(
                        License()
                            .name("Propriétaire")
                            .url("https://investpro.ma")
                    )
            )
            .addSecurityItem(SecurityRequirement().addList(securitySchemeName))
            .components(
                io.swagger.v3.oas.models.Components()
                    .addSecuritySchemes(
                        securitySchemeName,
                        SecurityScheme()
                            .name(securitySchemeName)
                            .type(SecurityScheme.Type.HTTP)
                            .scheme("bearer")
                            .bearerFormat("JWT")
                    )
            )
    }
}
